WordPress/src/main/java/org/wordpress/android/ui/jetpack/scan/ScanViewModel.kt

```diff
@@ -71,15 +72,29 @@ class ScanViewModel @Inject constructor(
         }
         isStarted = true
         this.site = site
-        fetchScanState()
+        init()
+    }
+
+    private fun init() {
+        launch {
+            scanStateModel = scanStore.getScanStateForSite(this@ScanViewModel.site)
+            scanStateModel?.let {
+                updateUiState(buildContentUiState(it))
+                if (fixableThreatIds.isNotEmpty()) fetchFixThreatsStatus(fixableThreatIds)
+            }
+            fetchScanState()
```

Comment:
❓ Question: Do we intentionally start fetching the scanState after the "fetchFixThreatsStatus" completes or could they run in parallel?

Reply:
Right now, I intentionally start fetching the scanState after the "fetchFixThreatsStatus" completes to avoid any conflict in the progress bar display logic for threats fixing state and scanning state (which could've been triggered from Calypso while threat are fixing).